backend/src/impl/default_controllers_impl.py

```diff
@@ -34,21 +34,38 @@
     BenchmarkConfig,
     DatasetMetadata,
     SingleAnalysis,
+    SystemOutput,
 )
 from explainaboard_web.models.datasets_return import DatasetsReturn
 from explainaboard_web.models.language_code import LanguageCode
 from explainaboard_web.models.system import System
 from explainaboard_web.models.system_analyses_return import SystemAnalysesReturn
 from explainaboard_web.models.system_create_props import SystemCreateProps
 from explainaboard_web.models.system_info import SystemInfo
-from explainaboard_web.models.system_output import SystemOutput
+from explainaboard_web.models.system_update_props import SystemUpdateProps
 from explainaboard_web.models.systems_analyses_body import SystemsAnalysesBody
 from explainaboard_web.models.systems_return import SystemsReturn
 from explainaboard_web.models.task import Task
 from explainaboard_web.models.task_category import TaskCategory
 from flask import current_app
 from pymongo import ASCENDING, DESCENDING
 
+
+def _is_creator(system: System, user: User) -> bool:
+    """check if a user is the creator of a system"""
+    return system.creator == user.email
+
+
+def _is_shared_user(system: System, user: User) -> bool:
+    """check if a user is a shared user of a system"""
+    return system.shared_users and user.email in system.shared_users
+
+
+def _has_read_access(system: System, user: User) -> bool:
```

Comment:
I think it'll make this helper function more useful if we add
```
user = get_user()
user.is_authenticated and _has_read_access(system, user)
```
to it. `get_user()` is cached so we can call it as many times as we want.

Reply:
Thanks, I went a step further and included `system.is_private`. Now we only need to call _has_read_access(system) or _has_write_access(system)